build.sbt

```diff
@@ -332,51 +332,22 @@ lazy val codegenScalaCompilationTest = project
     ReflectiveCodeGen.copyUnmanagedSources := false)
 
 lazy val codegenJavaCompilationExampleSuite: CompositeProject =
-  if (sys.props.contains("example.suite.java.enabled")) {
-    new ExampleSuiteCompilationProject {
-      val name = "codegenJavaCompilationExampleSuite"
-      val pathToTests = "codegen/java-gen/src/test/resources/tests"
-      val innerProjects = findProjects.map { case (dir, name) =>
-        Project("test-java" + name, dir)
-          .dependsOn(sdkJava % "compile", sdkJavaTestKit % "test")
-          .disablePlugins(HeaderPlugin, JavaFormatterPlugin)
-          .settings(
-            Compile / unmanagedSourceDirectories ++= Seq("generated-managed", "generated-unmanaged").map(
-              baseDirectory.value / _),
-            Test / unmanagedSourceDirectories ++= Seq("generated-test-managed", "generated-test-unmanaged").map(
-              baseDirectory.value / _),
-            Compile / PB.protoSources += baseDirectory.value / "proto",
-            akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Java))
-          .enablePlugins(ProtocPlugin, AkkaGrpcPlugin)
-      }
-    }
-  } else {
-    ExampleSuiteCompilationProject.empty
+  ExampleSuiteCompilationProject.compilationProject("java", "codegen/java-gen/src/test/resources/tests") {
+    testProject =>
+      testProject
+        .dependsOn(sdkJava % "compile", sdkJavaTestKit % "test")
+        .settings(akkaGrpcGeneratedLanguages := Seq(AkkaGrpc.Java))
```

Comment:
to build a compilation project, we need three things: 

* the lang so we can use it as label to name / prefix things
* the path to the tests - (I didn't want to do concatenation on this to make it clear in the build.sbt file which path we are using
* a function `Project => Project` to configure the project with some specifics